codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/auth/AuthSchemeResolverGenerator.kt

```diff
@@ -206,4 +211,33 @@ class AuthSchemeResolverGenerator(
             )
         }
     }
+
+    private fun noAuthSchemeOptionResolver() =
+        writable {
+            // For encapsulation, we do not expose `NoAuthSchemeResolver`, instead expose a creation function
+            // that returns an opaque type.
+            rustTemplate(
+                """
+                ##[derive(Debug)]
+                struct NoAuthSchemeResolver;
+                impl ResolveAuthScheme for NoAuthSchemeResolver {
+                    fn resolve_auth_scheme<'a>(
+                        &'a self,
+                        _params: &'a #{Params},
+                        _cfg: &'a #{ConfigBag},
+                        _runtime_components: &'a #{RuntimeComponents},
+                    ) -> #{AuthSchemeOptionsFuture}<'a> {
+                        #{AuthSchemeOptionsFuture}::ready(#{Ok}(vec![#{NoAuthSchemeOption:W}]))
+                    }
+                }
+
+                /// Return an auth scheme resolver that favors `noAuth`
+                pub fn no_auth_scheme_resolver() -> impl ResolveAuthScheme {
```

Comment:
I guess it comes down to I'm not a fan of this. This is a niche use case and I'm not sure it's useful to the majority of customers that we should include it by default. If we need it for protocol tests then maybe we just place it there instead.

Reply:
We can. Haven't we had some number of customers requesting disabling signing, though (this convenience resolver was brought up in our team design doc, which is why I originally put it here)? Maybe do we ask them to override sigv4 auth scheme with no-signer by themselves instead of providing this resolver?

Reply:
> Haven't we had some number of customers requesting disabling signing, though

We have though I think many times it was because they wanted to use sigv4 OR something else in which case they need to disable signing and a custom interceptor implemented their auth logic. That would no longer be necessary with the ability to register a custom auth scheme though.

Reply:
Addressed in c94eb95